File: 1. data base connection program/database conectivity/python/mysqlone.py
# program 1: 
# Required Library for mysql connection to python:
# 1. pip install mysql-connector-python

import mysql.connector

# Connect to MySQL
conn = mysql.connector.connect(
    host="localhost",
    user="yourusername",
    password="yourpassword",
    database="yourdatabase"
)

cursor = conn.cursor()

# Example: Create table
cursor.execute("CREATE TABLE IF NOT EXISTS users (id INT AUTO_INCREMENT PRIMARY KEY, name VARCHAR(255), age INT)")

# Insert
cursor.execute("INSERT INTO users (name, age) VALUES (%s, %s)", ("Bob", 30))
conn.commit()

# Fetch
cursor.execute("SELECT * FROM users")
for row in cursor.fetchall():
    print(row)

conn.close()


# Program 2: 
# Required Library for postgresql connection to python:
# 1.    pip install psycopg2-binary


import psycopg2

# Connect to PostgreSQL
conn = psycopg2.connect(
    host="localhost",
    database="yourdatabase",
    user="yourusername",
    password="yourpassword"
)

cursor = conn.cursor()

# Example: Create table
cursor.execute("CREATE TABLE IF NOT EXISTS users (id SERIAL PRIMARY KEY, name VARCHAR(100), age INT)")

# Insert
cursor.execute("INSERT INTO users (name, age) VALUES (%s, %s)", ("Charlie", 35))
conn.commit()

# Fetch
cursor.execute("SELECT * FROM users")
print(cursor.fetchall())

conn.close()

File: 1. data base connection program/database conectivity/java/mysqlone.java
import java.sql.*;

public class MySQLConnection {
    public static void main(String[] args) {
        String url = "jdbc:mysql://localhost:3306/yourdatabase";
        String user = "yourusername";
        String password = "yourpassword";

        try {
            // Load JDBC driver (optional in modern versions)
            Class.forName("com.mysql.cj.jdbc.Driver");

            // Connect to database
            Connection conn = DriverManager.getConnection(url, user, password);

            // Create statement
            Statement stmt = conn.createStatement();

            // Execute query
            ResultSet rs = stmt.executeQuery("SELECT * FROM users");

            // Print results
            while (rs.next()) {
                System.out.println(rs.getInt("id") + ": " + rs.getString("name") + " - " + rs.getInt("age"));
            }

            // Close
            rs.close();
            stmt.close();
            conn.close();

        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}



// | Database       | JDBC URL Example                                    | Driver Class Name                              |
// | -------------- | --------------------------------------------------- | ---------------------------------------------- |
// | **SQLite**     | `jdbc:sqlite:mydatabase.db`                         | `org.sqlite.JDBC`                              |
// | **PostgreSQL** | `jdbc:postgresql://localhost:5432/mydatabase`       | `org.postgresql.Driver`                        |
// | **SQL Server** | `jdbc:sqlserver://localhost:1433;databaseName=MyDB` | `com.microsoft.sqlserver.jdbc.SQLServerDriver` |

